tests/kubesonic/test_k8s_join_disjoin.py

```diff
@@ -0,0 +1,387 @@
+        minikube start \
+        --listen-address=0.0.0.0 \
+        --apiserver-port=6443 \
+        --ports=6443:6443 \
+        --extra-config=kubeadm.skip-phases=addon/kube-proxy,addon/coredns \
+        --install-addons=false \
+        --kubernetes-version={KUBERNETES_VERSION} \
+        --apiserver-ips={vmhost.mgmt_ip} \
+        --force \
+    '''
+    vmhost.shell(k8s_master_setup_cmd)
+    logger.info("K8s master setup is done")
+
+
+def remove_k8s_master(vmhost):
+    logger.info("Start to remove k8s master on vmhost")
+    vmhost.shell("minikube delete --all --purge")
+    vmhost.shell("rm -f /tmp/juju-mk*")
+    vmhost.shell("rm -f /tmp/minikube*")
+    logger.info("K8s master is removed")
+
+
+# Minikube's default pki ca path is /var/lib/minikube/certs/ca.crt
+# But DUT's default pki ca path is /etc/kubernetes/pki/ca.crt, so need update
+def update_kubelet_config(vmhost, creds):
+    logger.info("Start to update kubelet config")
+    http_proxy = creds.get("proxy_env", {}).get("http_proxy", "")
+    https_proxy = creds.get("proxy_env", {}).get("https_proxy", "")
+    http_proxy_param = f"http_proxy={http_proxy}" if http_proxy else ""
+    https_proxy_param = f"https_proxy={https_proxy}" if https_proxy else ""
+    proxy_param = f"{http_proxy_param} {https_proxy_param} {NO_PROXY}"
+    tmp_kubelet_config = "/tmp/kubelet-config.yaml"
+    get_kubelet_config_cmd = f"{proxy_param} minikube kubectl -- get cm {KUBELET_CONFIGMAP} -n kube-system -o yaml"
+    vmhost.shell(f"{get_kubelet_config_cmd} > {tmp_kubelet_config}")
+    vmhost.shell(f"sed 's|/var/lib/minikube/certs/ca.crt|/etc/kubernetes/pki/ca.crt|' -i {tmp_kubelet_config}")
+    vmhost.shell(f"{NO_PROXY} minikube kubectl -- apply -f {tmp_kubelet_config}")
+    vmhost.shell(f"rm -f {tmp_kubelet_config}")
+    logger.info("Kubelet config is updated")
+
+
+# Minikube kubectl need to param
+def update_vmhost_param(vmhost):
+    logger.info("Start to update vmhost param")
+    global VMHOST_PARAM_DEFAULT
+    param_default_value = vmhost.shell("sysctl fs.protected_regular", module_ignore_errors=True)["stdout"]
+    if param_default_value:
+        VMHOST_PARAM_DEFAULT = param_default_value.replace(" ", "")
+    vmhost.shell("sysctl fs.protected_regular=0")
+    logger.info("Vmhost param are updated")
+
+
+def restore_vmhost_param(vmhost):
+    logger.info("Start to restore vmhost param")
+    if VMHOST_PARAM_DEFAULT:
+        vmhost.shell(f"sysctl {VMHOST_PARAM_DEFAULT}")
+    logger.info("VMhost param are restored")
+
+
+def check_k8s_state_db(duthost):
+    logger.info("Start to check k8s state db")
+    get_update_time_cmd = "sonic-db-cli STATE_DB hget 'KUBERNETES_MASTER|SERVER' update_time"
+    update_time = duthost.shell(f"{get_update_time_cmd}", module_ignore_errors=True)["stdout"]
+    duthost.shell("systemctl status ctrmgrd")
+    if not update_time:
+        duthost.shell("sonic-db-cli STATE_DB hset 'KUBERNETES_MASTER|SERVER' update_time '2024-12-24 01:01:01'")
```

Comment:
Why do we hard code update_time?

Reply:
Ctrmgrd service just needs that the update_time column has a value, otherwise it needs to wait for extra 5 minutes to join. It's the code logic for device boot case.